.github/workflows/github-actions.yml

```diff
@@ -0,0 +1,59 @@
+name: Tests
+on:
+  push:
+    branches: '**'
+  pull_request:
+    branches: main
+
+jobs:
+  unit-testing:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+    
+    steps:
+      - name: Install xmllint on Linux
+        if: runner.os == 'Linux'
+        run: sudo apt-get install libxml2-utils
+        shell: bash
+      
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+      
+      - name: Set environment variables
+        run:  echo "PYTHON_VERSION=$(./python_version.sh)" >> $GITHUB_ENV
```

Comment:
I question whether you need to have `pythn_version.sh` and, instead, do:

```
echo "PYTHON_VERSION=$(curl --location https://devguide.python.org/versions | xmllint --html --xpath '//section[@id="supported-versions"]//table/tbody/tr[count(//section[@id="supported-versions"]//table/tbody/tr[td[.="security"]]/preceding-sibling::*)]/td[1]/p/text()' - 2> /dev/null)" >> $GITHUB_ENV
```

Reply:
Not really. I wanted to avoid spreading the "ugliness" of curl function. But I can see there are two costs associated with it: (i) adding an extra file, and (ii) forcing a reader of Github-Action workflow to look away to see what `pythn_version.sh` does.

Reply:
Exactly!